botnim/query.py

```diff
@@ -111,20 +112,97 @@ def get_index_mapping(self) -> Dict:
             logger.error(f"Failed to get index mapping: {str(e)}")
             raise
 
-def run_query(query_text: str, environment: str, bot_name: str, context_name: str, num_results: int = 7) -> List[SearchResult]:
+    def get_default_num_results(self) -> int:
+        """
+        Get the default number of results for the current context
+        
+        Returns:
+            int: Default number of results
+        """
+        # Find the specific context configuration
+        context_config = next(
+            (ctx for ctx in self.config.get('context', []) if ctx['name'] == self.context_name),
+            {}
+        )
+        
+        # Get default num_results from context config or use a reasonable default
+        return context_config.get('default_num_results', 7)
+
+    def query(self, query_text: str, num_results: int = None) -> List[SearchResult]:
+        """
+        Query the vector store with the given text
+        
+        Args:
+            query_text (str): The text to search for
+            num_results (int, optional): Number of results to return, or None to use context default
+        
+        Returns:
+            List[SearchResult]: List of search results
+        """
+        if num_results is None:
+            # Use the method to get default num_results
+            num_results = self.get_default_num_results()
+        
+        return self.search(query_text, num_results)
+
+def run_query(environment: str, bot_name: str, context_name: str, query: str, num_results: int = None, format: str = "dict"):
```

Comment:
Changing the parameter order for an existing function requires you to make sure that all users of this function are updated.

Reply:
in a way it is a left over of a previous version (which you asked me to fix). The only usage was to better deal with num_results. I moved it into the search function.

Reply:
oh... very bad, I wonder how did it work for me locally after this change. reverted.